src/main/java/com/example/board/global/validator/PostValidator.java

```diff
@@ -0,0 +1,24 @@
+package com.example.board.global.validator;
+
+import com.example.board.domain.post.entity.Post;
+import org.springframework.util.Assert;
+
+import java.util.Optional;
+
+public class PostValidator {
+
+    public static Post validateOptionalPostExists(Optional<Post> mayBePost) {
```

Comment:
Optional을 파라미터로 사용되는건 지양해야 하는 내용이긴 하지만,
의도상은 이해했습니다.

entity에 대한 정보가 없는 것에 대한 예외처리를 하려는 것으로 파악을 했는데요.
비즈니스 적인 부분이라기보다는 Jpa라는 라이브러리에 제약된 내용으로 보여져 아래와 같은 내용으로 util 클래스를 만드는게 어떠신가요?
```suggestion
public class EntityValidator<T> {
	
	public T validateOptionalExists(Optional<T> optional) {
		if (optional.isEmpty()) {
			throw new EntityNotFoundException("exception.entity: {}" + optional.getClass());
		}
		return optional.get();
	}
}
```

Reply:
<<수정했습니다>>

공통되는 Optional 검증을 작성하여 중복되는 기능을 없앴습니다.

[Optional 엔티티 검증 기능 리팩터링](https://github.com/hyena0608/springboot-board-jpa/commit/5f40e208a49a826b4d6eed6414af6aeea464f2df)